src/scripts/actions/UserActions.js

```diff
@@ -4,27 +4,97 @@ import { createAction } from 'redux-actions'
 import Promise from 'bluebird'
 import Cookies from 'js-cookie'
 
-import { LOGIN_REQUESTED, LOGIN_SUCCESS, LOGOUT } from 'constants/ActionConstants'
+import {
+  LOGIN_REQUESTED,
+  LOGIN_SUCCESS,
+  LOGOUT,
+  SET_WALLET_DATA
+} from 'constants/ActionConstants'
+import paratii from 'utils/ParatiiLib'
+import { DEFAULT_PASSWORD } from 'constants/ParatiiLibConstants'
+import { getWalletKey, getMnemonicKey } from 'selectors/index'
 
-import type { Dispatch } from 'redux'
+import type { RootState, Dispatch } from 'types/ApplicationTypes'
 
 const loginRequested = createAction(LOGIN_REQUESTED)
 const loginSuccess = createAction(LOGIN_SUCCESS)
 const logoutAction = createAction(LOGOUT)
+const setWalletData = createAction(SET_WALLET_DATA)
 
-const sleep = (ms) => {
+const sleep = ms => {
   return new Promise(resolve => setTimeout(resolve, ms))
 }
 
-export const login = (email: string, password: string) => (dispatch: Dispatch<*>) => {
+export const login = (email: string, password: string) => (
+  dispatch: Dispatch
+) => {
   dispatch(loginRequested())
   sleep(200).then(() => {
     Cookies.set('email', email)
-    dispatch(loginSuccess({email}))
+    dispatch(loginSuccess({ email }))
   })
 }
 
-export const logout = () => (dispatch: Dispatch<*>) => {
-  dispatch(logoutAction())
+export const logout = () => (dispatch: Dispatch) => {
   Cookies.remove('email')
+  dispatch(logoutAction())
+}
+
+const setAndSyncWalletData = ({
+  walletKey,
+  wallet,
+  mnemonic,
+  mnemonicKey
+}: {
+  walletKey: string,
+  wallet: Object,
+  mnemonicKey: string,
+  mnemonic: string
+}) => (dispatch: Dispatch): void => {
+  dispatch(
+    setWalletData({
+      walletKey,
+      mnemonicKey
+    })
+  )
+  localStorage.setItem(walletKey, JSON.stringify(wallet))
+
+  // TODO: needs to be encrypted
+  localStorage.setItem(mnemonicKey, mnemonic)
+}
+
+export const setupKeystore = () => async (
+  dispatch: Dispatch,
+  getState: () => RootState
+) => {
+  paratii.eth.wallet.clear()
+  const walletKey: string = getWalletKey(getState())
```

Comment:
Although refactored, the app still always starts w/ a default key of `keystore-anon`. In reality, though, if the user has secured their account we need some way to get their last wallet key.

However, given how it is currently stored, (where `walletKey` is the actual localStorage key), I'm not sure that would be possible.

Thoughts, @jellegerbrandy, on storing all of our paratii data in `localStorage` in this format?

`localStorage.setItem('paratii', JSON.stringify({ walletKey: 'keystore-anon', wallet: { data: value, key: thing }, //etc }))`

Then on app initialization we can `JSON.parse(localStorage.getItem('paratii'))` one time and put the data in the store. Our defaults will still remain `keystore-anon` and `mnemonic-anon`